core/controlplane/config/templates/cloud-config-etcd

```diff
@@ -130,34 +134,202 @@ ssh_authorized_keys:
 
 write_files:
 
+  - path: /opt/bin/cfn-init-etcd-server
+    owner: root:root
+    permissions: 0700
+    content: |
+      #!/bin/bash -vxe
+
+      cfn-init -v -c "etcd-server" --region {{.Region}} --resource {{.Etcd.LogicalName}}${{.EtcdIndexEnvVarName}} --stack ${{.StackNameEnvVarName}}
+
+  - path: /opt/bin/attach-etcd-volume
+    owner: root:root
+    permissions: 0700
+    content: |
+      #!/bin/bash -vxe
+
+      instance_id=$(curl http://169.254.169.254/latest/meta-data/instance-id)
+      private_ip=$(curl http://169.254.169.254/latest/meta-data/local-ipv4)
+      az=$(curl http://169.254.169.254/latest/meta-data/placement/availability-zone)
+      stack_name=$(
+        aws ec2 describe-tags --region {{.Region}} --filters \
+          "Name=resource-id,Values=$instance_id" \
+          "Name=key,Values=aws:cloudformation:stack-name" \
+          --output json \
+        | jq -r ".Tags[].Value"
+      )
```

Comment:
nit: stack_name should be taken from the env var `KUBE_AWS_STACK_NAME` injected via the embedded userdata in stack-template.json because it is a more portable way to get the stack name. More concretely, the `aws:cloudformation:stack-name` and other tags won't be populated when the EC2 instance is created via a spot fleet.